Elaborate on release process documentation (#1251)
open-component-model/ocm: RELEASE_PROCESS.md

creation of a "Release Candidate" which can be tested
and delivered to end users willing to test the new release.

### Schedule

Minor releases are done every sprint, i.e. every two weeks. To be even more precise,
technically speaking, two releases are produced every sprint:

* Release Candidate for the current version in main

> * Promotion of the previous sprint's candidate to a normal minor version

Reviewer: ```suggestion
* Promotion of the previous sprint's candidate to an official minor version
```
